Fix null drops

gchpaco opened the pull request:
Probably fixes #33.  Certainly fixes https://github.com/Vazkii/Quark/issues/1345 and cousins.

I can reproduce with Quark oddities with Totem of Holding on, and ToolBelt 1.9.6.

gchpaco commented:
As discussed in #33 we still have a problem here even with this patch.  Am unsure what is going on right now, but it's definitely something that changed between 1.9.0 and 1.9.6.

gigaherz commented on src/main/java/gigaherz/toolbelt/slot/ExtensionSlotBelt.java:
```diff
@@ -173,7 +173,10 @@ public void playerDeath(PlayerDropsEvent event)
             {
-                event.getDrops().add(player.dropItem(instance.getBelt().getContents(), true, false));
+                ItemStack belt = instance.getBelt().getContents();
+                if (belt != null) {
```
If getContents returns null, it should be fixed to never return null, the null check here shouldn't be necessary.

gigaherz commented on src/main/java/gigaherz/toolbelt/slot/ExtensionSlotBelt.java:
```diff
@@ -186,7 +189,7 @@ public void playerClone(PlayerEvent.Clone event)
             if (oldBelt == null) return;
             ItemStack item = oldBelt.getBelt().getContents();
-            if (newBelt == null) {
+            if (newBelt == null && item != null) {
```
Same as before, getContents should never ever be returning null, so that needs to be fixed instead, this change shouldn't be necessary.

gigaherz commented on build.gradle:
```diff
@@ -52,6 +55,8 @@ dependencies {
     deobfCompile "mezz.jei:jei_1.12.2:4.15.0.281:api"
     runtime "mezz.jei:jei_1.12.2:4.15.0.281"
 
+    deobfCompile "com.azanor:Baubles:1.12-1.5.2"
```
👌 I was originally using baubles as a .jar file instead of a maven dependency, so I never remembered to change this.

gchpaco commented:
Addressed your comments, but I'm still getting this trouble, so we're not out of the woods yet.
[crash-2019-06-29_16.50.11-server.txt](https://github.com/gigaherz/ToolBelt/files/3342336/crash-2019-06-29_16.50.11-server.txt)

gigaherz commented on src/main/java/gigaherz/toolbelt/customslots/ExtensionSlotItemHandler.java:
```diff
@@ -42,7 +42,8 @@ public ResourceLocation getType()
     {
-        return inventory.getStackInSlot(slot);
+        ItemStack result = inventory.getStackInSlot(slot);
+        return result == null ? ItemStack.EMPTY : result;
```
Err, this is my fault I guess. I should have been more clear: the situation where *a slot contains null* should never happen, so this still shouldn't be needed. :P
We need to track down which part of the code is setting the slot to null, because that part of the code is at fault.

gchpaco replied:
Oh okay!

gchpaco replied:
I should note that it's happening on a fresh life, i.e. I suspect that the slot is being initialized to null.

gigaherz replied:
Well, I'm using forge's ItemStackHandler, which uses a NonNullList set to get initialized to ItemStack.EMPTY.... so that's unlikely.

I'm gonna add quark locally and try it myself, see if I can catch who's setting things to null.

duely replied:
Yeah, the ItemStack contract is that null will never be returned, but from what I understood of the code associated with that Quark bug (which was actually fixed in a patch) and [this](https://github.com/Vazkii/Botania/issues/2939) issue with Botania (and an adjacent issue with Electroblob's Wizardry), it's [this](https://github.com/Vazkii/Botania/blob/master/src/main/java/vazkii/botania/common/item/ItemKeepIvy.java#L44) line that's problematic -- a null being inserted into the EntityItem list. I haven't had a chance to actually look through gchpaco's PR to see what's being addressed to see how this directly relates to the EntityItems.

gigaherz replied:
PROBLEM FOUND! I forgot to add a check to make sure the slot is not empty.
dropItem should only ever be called if the slot contains a non-empty stack!

Thanks for the attempt @gchpaco, I can take it from here, it's just literally two lines that I already fixed locally while trying things. :P

gigaherz commented:
Merged the relevant part of this (the build.gradle edits) in 82bf02a7e25
